File: app/src/test/java/com/example/envelopemoney/HistoryTransferTotalsTest.java
package com.example.envelopemoney;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class HistoryTransferTotalsTest {

    @Test
    public void inboundMirrorIsExcludedFromHistoryTotal() {
        assertFalse(HistoryTransferTotals.includeAmountInHistoryTotal(true, false));
        assertTrue(HistoryTransferTotals.includeAmountInHistoryTotal(true, true));
        assertTrue(HistoryTransferTotals.includeAmountInHistoryTotal(false, false));
    }

    @Test
    public void spendingTotalIgnoresInboundAndKeepsOutgoingOutWhenTransfersVisible() {
        // $20 spend + $50 source transfer; $40 allocated out; inbound mirror not in spendGross.
        assertEquals(30.0d, HistoryTransferTotals.spendingTotal(70.0d, 40.0d, true), 0.0001d);
        assertEquals(70.0d, HistoryTransferTotals.spendingTotal(70.0d, 40.0d, false), 0.0001d);
    }

    @Test
    public void inboundPanelStaysNegativeWhenDestinationIsSelected() {
        double panel = 0d;
        panel = HistoryTransferTotals.addInboundToPanel(panel, 40.0d);
        panel = HistoryTransferTotals.addInboundToPanel(panel, 10.0d);
        assertEquals(-50.0d, panel, 0.0001d);
        assertEquals("To Savings: -$50.00", HistoryTransferTotals.formatToSummary("Savings", panel));
    }
}

File: web/test/historyTransferTotals.test.js
import { test } from 'node:test';
import assert from 'node:assert/strict';
import {
  includeAmountInHistoryTotal,
  spendingTotal,
  addInboundToPanel,
  formatToSummary,
} from '../domain/historyTransferTotals.js';

test('inbound mirror is excluded from history total', () => {
  assert.equal(includeAmountInHistoryTotal(true, false), false);
  assert.equal(includeAmountInHistoryTotal(true, true), true);
  assert.equal(includeAmountInHistoryTotal(false, false), true);
});

test('spending total ignores inbound and keeps outgoing out when transfers visible', () => {
  assert.equal(spendingTotal(70, 40, true), 30);
  assert.equal(spendingTotal(70, 40, false), 70);
});

test('inbound panel stays negative when destination is selected', () => {
  let panel = 0;
  panel = addInboundToPanel(panel, 40);
  panel = addInboundToPanel(panel, 10);
  assert.equal(panel, -50);
  assert.equal(formatToSummary('Savings', panel), 'To Savings: -$50.00');
});
